### ISSUE TITLE
Users information leak reported by UNESCO

### ISSUE THREAD

**Issue by giohappy** [geosolutions-it/unesco-ihp#324]
UNESCO reported that GeoNode's **/owners API endpoint return the full list of users, including ther personal information (!!!!)**.
Of course it must be fixed, and any other APi exposing users info must be locked down.

Only administrators must be able to retrieve such information.
The new APi (v2) must be strict on this side, assuming everywhere a "private by default" approach.

**Comment by giohappy** [comment]
The list of owners is used by the frontend to provide the filtering by "author" (it would be to rename the endpoint and any reference like this), so if we hide them totally it will break this functionality.

My proposal is to:

1. list only fiest name and last name (only admins can view the full list of attributes)
2. for anonymous users: list only owners which have at least one public resource
3. for registered users: list only the owners of resources that the user have permissions to view

**Comment by afabiani** [comment]
@giohappy any progress here? How should I proceed?

**Comment by giohappy** [comment]
we're waiting for a feedback from UNESCO to verify that the proposed solution suffice.

**Comment by marthamareal** [comment]
> The list of owners is used by the frontend to provide the filtering by "author" (it would be to rename the endpoint and any reference like this), so if we hide them totally it will break this functionality.
> 
> My proposal is to:
> 
> 1. list only first name and last name (only admins can view the full list of attributes)
> 2. only return owners having a resource shared with the current authenticated user
> 
> The last point means that anonymous users will only see owners having at least one public resource. Registered users will see owners of resources either shared with them on a group they are member of.

@giohappy  the search url uses username, if hidden then other users will not be able to filter resources by usernames.

**Comment by giohappy** [comment]
@marthamareal you're right. And probably usernme is enoguh. We could exclude first name and last nams to non-admin users.

It's important (for non admins) to only return the usernames for the owners of the resources (and only the resources) the logged in user has permission to read.

**Comment by giohappy** [comment]
fixed through https://github.com/GeoNode/geonode/pull/7295
